### PR TITLE
Fix GitHub CI/CD

### PR DESCRIPTION
**Pull request by keyonjerome**
PR starting as draft to make sure they run before merging.

### CODE REVIEW

**Comment by keyonjerome** [comment]
PR working. Requesting @niiquaye's review, but pinging @ColinToft @Nick-palmar @meshvaD for visibility:
A few notes:
- The version of `flake8` in our image seems not to support the Python Walrus operator, so I removed it from drivetrain's `launch.py`. 
- Upgrading ament-flake8, like we'd done before, seemed to result only in more, less traceable issues, which were making all PR builds fail in the first place. I don't think it's worth investigating further, so let's stick with the Python version given in the image if we can for now (3.8 iirc). If you're interested, I've attached the failing test below.
- I've removed Windows support. The issue with Windows support on our build seemed to lie with CMake headers in the `rcutils` package coming from `control_msgs`, and given that Galactic is `EOL`, I doubt we'd have a shot at getting any fixes backported anyways. If we ever start using Windows as a fully supported development environment for all of our team members, it might be worth revisiting, but not now. Once again, I'll attach the failing test.

flake8 test:
[failingamenttest.txt](https://github.com/uwrobotics/uwrt_mars_rover/files/10310955/failingamenttest.txt)

rcutils/control-msgs test:
[failing-rcutils-control-msgs-test.txt](https://github.com/uwrobotics/uwrt_mars_rover/files/10310950/failing-rcutils-control-msgs-test.txt)

File: windows_amd64_upstream_dependencies.repos [hunk 1]
```diff
@@ -19,8 +19,8 @@ repositories:
     version: galactic
   xacro:
     type: git
-    url: https://github.com/wmmc88/xacro.git # FORK. TODO: switch back to ros2 branch on upstream once https://github.com/ros/xacro/pull/304 merges and releases
```
Comment (Nick-palmar) [inline windows_amd64_upstream_dependencies.repos:22]: I see that the linked PR is merged so I assume changing back if fine but maybe this breaks something else down the line? Might be worth leaving a comment in there in case

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -29,12 +32,12 @@ jobs:
         run: echo "::set-output name=image::${{ env.IMAGE }}"
 
   linters:
+    if: ${{ !github.event.act }}
     name: ament-${{ matrix.linter }}
     needs: prepare-actions-environment
     runs-on: ubuntu-latest
     container:
       image: ${{ needs.prepare-actions-environment.outputs.image }}
-      options: -u root
```
Comment (Nick-palmar) [inline .github/workflows/ci.yaml:37]: just curious, why was this removed?

Comment (keyonjerome) [reply .github/workflows/ci.yaml]: Oh this was me messing around to see if `rosdep update` was causing the issue. `rosdep` does not like being run as root, but that was not the issue nor do I think this line of code touches `rosdep`; readded. Nice catch.

File: windows_amd64_upstream_dependencies.repos (hunk 1 above)
Comment (keyonjerome) [reply windows_amd64_upstream_dependencies.repos:22]: Leaving a comment in there

**Review by ColinToft — APPROVED** [review]
